### PR TITLE
Implement recommendations from "Woo Quality Insights"

### PR DESCRIPTION
**Pull request by mntzrr**
https://app.clickup.com/t/8694vyr9u

### CODE REVIEW

File: classes/requests/class-kp-requests.php [hunk 1]
```diff
@@ -99,7 +99,7 @@ public function set_credentials() {
 	}
 
 	public function calculate_auth() {
-		return 'Basic ' . base64_encode( $this->merchant_id . ':' . htmlspecialchars_decode( $this->shared_secret ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions -- Base64 used to calculate auth headers.
+		return 'Basic ' . base64_encode( $this->merchant_id . ':' . htmlspecialchars_decode( $this->shared_secret, ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401 ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions -- Base64 used to calculate auth headers.
```
Comment (MichaelBengtsson) [inline classes/requests/class-kp-requests.php:102]: `htmlspecialchars_decode` default flags are already  `ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401`. So why is that needed here?
https://www.php.net/manual/en/function.htmlspecialchars-decode.php

Comment (mntzrr) [reply classes/requests/class-kp-requests.php:102]: That's correct, however it is required to be explicitly set.

image

Comment (MichaelBengtsson) [reply classes/requests/class-kp-requests.php:102]: Ah ok, then i see. Thanks for the clarification
